add version 1 of dynamic rule (#2334)
port-labs/port-docs: docs/search-and-query/search-and-query.md

]
```

> ### contextual query rules

Reviewer: ```suggestion
### Contextual query rules
```

:::info Closed beta feature

> This capability is currently in closed beta, and is not yet generally available.

Reviewer: ```suggestion
This capability is currently in closed beta, and is not yet generally available.  
```

If you would like to join the beta, please reach out to us.
:::

> To filter and costomise queries, you can add the context of the user trigring the query, You can use contextual query rules with other rules as well to fully costomise the response.

Reviewer: ```suggestion
To implement specific and/or complex queries, you can add the context of the triggering user to a query rule, allowing you to access that user's entity and/or owning teams.  

You can mix contextual query rules freely with other rules as part of your queries.

This can be used in either the `property` or `value` key in a query rule:
```

Reviewer: move the "property & value" tabs here instead of under the next header
